lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -76,6 +76,20 @@ SmallVector<Direction> direction::unpackAttribute(Operation *component) {
 // Utilities
 //===----------------------------------------------------------------------===//
 
+/// Returns whether this value is either (1) a port on a ComponentOp or (2) a
+/// port on a cell interface.
+static bool isPort(Value value) {
+  Operation *definingOp = value.getDefiningOp();
+  return value.isa<BlockArgument>() ||
+         (definingOp && isa<CellInterface>(definingOp));
+}
+
+/// Gets the port for a given BlockArgument.
+ComponentPortInfo calyx::getComponentPortInfo(BlockArgument arg) {
+  Operation *op = arg.getOwner()->getParentOp();
+  return getComponentPortInfo(op)[arg.getArgNumber()];
```

Comment:
Unless someone know what's going on, it might not be obvious that the `op` now refers to what we expect to be a ComponentOp.  I think a change of the signature of the prior `getComponentPortInfo(ComponentOp)` would be sufficient to indicate what is going on.

Reply:
Going to address in #1733